src/core/qgsfeatureiterator.h

```diff
@@ -83,6 +83,14 @@ class CORE_EXPORT QgsAbstractFeatureIterator
       return mValid;
     }
 
+    /**
+     * Indicator if there was an error when sending the compiled query to the server.
+     * This indicates that there is something wrong with the expression compiler.
+     *
+     * \since QGIS 3.2
+     */
+    bool compileFailed() const;
```

Comment:
We've already got a method for this - see compileStatus

Reply:
I know. The difference is this one reflects errors and this one is (at least on some providers) only set after starting to iterate. Would you merge the two?